server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/MirroringServicePluginConfig.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.internal.mirror;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+final class MirroringServicePluginConfig {
```

Comment:
What do you think of making `PluginConfig` an interface and extending `PluginConfig` to provide mirror configuration as we did in `MirrorCredential`?

With this approach, users could set plugin configuration type-safely when using builder methods. For example:
```java
new CentralDogmaBuilder(...)
   . pluginConfigs(new MirrorPluginConfig(...), 
                   ControlPlanePluginConfig.disabled())
```

Reply:
Addressed. 👍